caproto/threading/client.py

```diff
@@ -1282,22 +1284,33 @@ def read(self, *args, timeout=2, **kwargs):
                 f"not respond to attempt to read channel named "
                 f"{self.name!r} within {timeout}-second timeout."
             )
+
+        # TODO shouldn't we get access to the ioid of our request?
```

Comment:
Good point. I think the approach was inherited from pyepics. We can and should be more precise about which reading we return.